luminoth/tasks.py

```diff
@@ -0,0 +1,157 @@
+"""Module where all generic task models are located.
+
+Each class corresponds to a different task, providing a task-specific API for
+ease of use. This API is common among different implementations, abstracting
+away the pecularities of each task model. Thus, no knowledge of the inner
+workings of said models should be needed to use any of these classes.
+"""
+from luminoth.tools.checkpoint import get_checkpoint_config
+from luminoth.utils.predicting import PredictorNetwork
+
+
+class Detector(object):
+    """Encapsulates an object detection model behavior.
+
+    In order to perform object detection with a model implemented within
+    Luminoth, this class should be used.
+
+    Attributes:
+        classes (list of str): Ordered class names for the detector.
+        prob (float): Default probability threshold for predictions.
+
+    TODO:
+        - Don't create a TF session internally (or make its creation optional)
```

Comment:
Could we re-use existing session if Luminoth is run via Python/Notebook?

Reply:
For notebooks, the problem is mainly the second point: internally `PredictorNetwork` always recreates the model, thus we end up with name clashes and suffixes. We must discard the old graph or find a way around that.

I haven't checked if there's a way to get the session from within. But in the case of TF-Eager, we simply don't want a session to exist (we'll probably have to check with `tf.executing_eagerly`).